galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistUtils.java

```diff
@@ -0,0 +1,214 @@
+						
+						runs.add(worklistItem);
+						
+					} catch (ResultArchiveStoreException e) {
+						throw new ServletException("Error while retriving Worklist information for Run ID " + runId, e);
+					}
+					
+				} else {
+					WorklistQuery.runIdsNotFound.add(runId);
+					try {
+						removeRunIdFromDss(runId);
+					} catch (Exception e) {
+						throw new ServletException("Error while updating the DSS", e);
+					}
+				}
+				
+			}
+		}
+		
+		return runs;
+	}
+	
+	protected IRunResult getRunById(String id) throws ResultArchiveStoreException {
+
+		IRunResult run = null;
+	       
+        for (IResultArchiveStoreDirectoryService directoryService : framework.getResultArchiveStore().getDirectoryServices()) {
+           
+           run = directoryService.getRunById(id);
+           
+           if(run != null) {
+              break;
+           }
+           
+        }
+		
+		return run;
+	}
+	
+	protected void removeRunIdFromDss(String runId) throws DynamicStatusStoreException, FrameworkPropertyFileException {
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String oldValue = dss.get("user.username.worklist");
+		
+		if (oldValue != null) {
+			
+			JsonObject jsonObject = new JsonParser().parse(oldValue).getAsJsonObject();
+			JsonArray runIds = (JsonArray) jsonObject.get("runIds");
+			
+			JsonElement id = new JsonParser().parse(runId);
+			if (runIds.contains(id)) {
+				runIds.remove(id);
+				jsonObject.remove("runIds");
+				jsonObject.add("runIds", runIds);
+			}	
+			
+			String newValue = jsonObject.toString();
+			
+			dss.performActions(
+					new DssSwap("user.username.worklist", oldValue, newValue));
+			
+		}
+			
+	}
+	
+	protected void addRunIdToDss(String runId) throws DynamicStatusStoreException, FrameworkPropertyFileException {
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String oldValue = dss.get("user.username.worklist");
+		
+		if (oldValue != null) {
+			
+			JsonObject jsonObject = new JsonParser().parse(oldValue).getAsJsonObject();
+			JsonArray runIds = (JsonArray) jsonObject.get("runIds");
+			
+			JsonElement id = new JsonParser().parse(runId);
+			if (!runIds.contains(id)) {
+				runIds.add(id);
+				jsonObject.remove("runIds");
+				jsonObject.add("runIds", runIds);	
+			}
+			
+			String newValue = jsonObject.toString();
+			
+			dss.performActions(
+					new DssSwap("user.username.worklist", oldValue, newValue));
+			
+		} else {
+			createWorklistPropertyInDss(runId);
+		}
+			
+	}
+	
+	protected void createWorklistPropertyInDss(String runId) throws DynamicStatusStoreException {
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		
+		JsonObject jsonObject = new JsonObject();
+		JsonArray runIds = new JsonArray();
+		runIds.add(runId);
+		jsonObject.add("runIds", runIds);
+		
+		String value = jsonObject.toString();
+		
+		dss.performActions(
```

Comment:
what happens if someone created the property as you were processing it,   need to consider this and do a swap instead,  I suggest you merge this with the swap code